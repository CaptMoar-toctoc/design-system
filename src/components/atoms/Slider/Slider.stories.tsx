import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Slider from './Slider';

export default {
	title: 'TocToc-DS/Atoms',
	component: Slider,
	// argTypes: {
	// 	checked:{control: 'boolean'}
	// },
} as ComponentMeta<typeof Slider>;

const Template: ComponentStory<typeof Slider> = (args) => <Slider {...args} /> ;

export const SliderTT = Template.bind({});


SliderTT.args = {
	disabled: false,
	defaultValue:50,
	handlerFn: () => { console.log(`click`) },
	customStyleObject: {},
};
SliderTT.decorators = [
	(Story) => (
	  <div style={{ maxWidth: '20%'}}>
		<Story />
	  </div>
	),
  ];
