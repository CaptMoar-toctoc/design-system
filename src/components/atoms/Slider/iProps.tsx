import { ReactNode } from "react";

export interface iProps {
    //children?: ReactNode;
    handlerFn?: any;
    disabled?: boolean,
    customStyleObject?: object,
	defaultValue: number
}