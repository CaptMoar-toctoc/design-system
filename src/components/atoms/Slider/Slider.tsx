import React, { FC } from 'react';
import { iProps } from './iProps';
import { ThemeProvider, Slider as SliderTT } from '@mui/material';

import theme from '../../theme/theme';

const Slider: FC<iProps> = ({
	handlerFn,
	disabled,
	customStyleObject,
	defaultValue
}) => {
	return (
			<ThemeProvider theme={theme}>

					<SliderTT
						valueLabelDisplay="auto"
						sx={customStyleObject}
						disabled={disabled}
						defaultValue={defaultValue}
					/>

			</ThemeProvider>
	);
}

export default Slider;