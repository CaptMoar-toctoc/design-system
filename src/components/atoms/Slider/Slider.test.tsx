import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';


import Slider from "./Slider";

describe('Input Testing Render', () => {
	//inicializacion de props componente
	//aca van todos los props requerido y OPCIONALES clea
	const props = {
		defaultValue: 25
	}

	test("Render SnapShot", () => {
		expect(render(<Slider {...props} />)).toMatchSnapshot();
	})


})

