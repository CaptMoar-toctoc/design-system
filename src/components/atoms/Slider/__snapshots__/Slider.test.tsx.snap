// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Testing Render Render SnapShot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-9xpw5l-MuiSlider-root"
      >
        <span
          class="MuiSlider-rail css-1svg07-MuiSlider-rail"
        />
        <span
          class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
          style="left: 0%; width: 25%;"
        />
        <span
          class="MuiSlider-thumb MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium css-1wfy9ex-MuiSlider-thumb"
          data-focusvisible="false"
          data-index="0"
          style="left: 25%;"
        >
          <input
            aria-orientation="horizontal"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="25"
            data-index="0"
            max="100"
            min="0"
            step="1"
            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
            type="range"
            value="25"
          />
          <span
            aria-hidden="true"
            class="MuiSlider-valueLabel css-1qhzk28-MuiSlider-valueLabel"
          >
            <span
              class="MuiSlider-valueLabelCircle"
            >
              <span
                class="MuiSlider-valueLabelLabel"
              >
                25
              </span>
            </span>
          </span>
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-9xpw5l-MuiSlider-root"
    >
      <span
        class="MuiSlider-rail css-1svg07-MuiSlider-rail"
      />
      <span
        class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
        style="left: 0%; width: 25%;"
      />
      <span
        class="MuiSlider-thumb MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium css-1wfy9ex-MuiSlider-thumb"
        data-focusvisible="false"
        data-index="0"
        style="left: 25%;"
      >
        <input
          aria-orientation="horizontal"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="25"
          data-index="0"
          max="100"
          min="0"
          step="1"
          style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
          type="range"
          value="25"
        />
        <span
          aria-hidden="true"
          class="MuiSlider-valueLabel css-1qhzk28-MuiSlider-valueLabel"
        >
          <span
            class="MuiSlider-valueLabelCircle"
          >
            <span
              class="MuiSlider-valueLabelLabel"
            >
              25
            </span>
          </span>
        </span>
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
